test/restart.jl

```diff
@@ -0,0 +1,73 @@
+@testset "Test restart with a different initial guess: $fun" for (fun, s) in ((:R2, :R2Solver), (:lbfgs, :LBFGSSolver), (:tron, :TronSolver), (:trunk, :TrunkSolver))
+  f(x) = (x[1] - 1)^2 + 4 * (x[2] - x[1]^2)^2
+  nlp = ADNLPModel(f, [-1.2; 1.0])
+
+  stats = GenericExecutionStats(nlp)
+  solver = eval(s)(nlp)
+  stats = SolverCore.solve!(solver, nlp, stats)
+  @test stats.status == :first_order
+  @test isapprox(stats.solution, [1.0; 1.0], atol = 1e-6)
+
+  nlp.meta.x0 .= 2.0
+  SolverCore.reset!(solver)
```

Comment:
Are you only able to reuse a solver when you change the starting point?
It's should work if you change the function but the Hessian / Jacobian of the new function have the same sparsity pattern.

Reply:
The idea is to call `reset!(solver)` if you change the initial guess, and `reset!(solver, nlp)` if you change the problem.
`Test restart with a different initial guess:` and `Test restart NLS with a different initial guess` are for the former and `Test restart with a different problem` and `Test restart NLS with a different problem` for the latter in the file `test/restart.jl`.

Reply:
Thanks for the clarification!